\section{Discussion}

In estimating the effect size of a gene-snp pair across many tissues, we have revealed new multivariate-patterns of heterogeneity and broadened our understanding of genetic effects beyond that of traditional `binary' analyses. Understanding such patterns may prove invaluable in assessing the genetic impact on multi-tissue phenotypes and grouping gene-snp pairs by similar multi-tissue patterns of heterogeneity, as well as in grouping tissues which tend to respond to genetic effects similarly. Our novel approach is the first of its kind to consider the continuous heterogeneity here evident, as well as to offer an analyses of a data-set of this size (44 tissues) as previous studies \cite{consortium_genotype-tissue_2015}. While the power improvement of joint analyses has been established \cite{flutre_statistical_2013} never before has such an analyses also considered quantitative heterogeneity among effects deemed `shared'. Furthermore, in developing a method for such analyses, it is possible to broaden its application to any assessment of multi-subgroup effects. The Likelihood-based training algorithm we describe allows our model the flexibility of adding and removing the appropriate  number of covariance matrices from the ultimate estimation.

Here, we present work on one-snp per gene, but as mentioned, armed with the hierarchical prior information from the overall data-set it is possible to infer a vector of estimated effects for any number of gene-snp pairs. 