\subsection{Methods Overview: Approach} 

We aim to learn about patterns of sharing across tissues within a SNP and among SNPs, which join to help us better understand the global and snp-specific patterns of effects of genetics on gene expression. 
This allows us to make comparisons among tissues in which the QTL is called active, and among gene-snp pairs with a similar degree of activity in a given tissue. 
Thus as an additional level of combining information, we assume that each eQTL may follow a particular pattern of activity characterized by its effects across tissues. Within these groups, the tissues exhibit characteristic patterns of sharing, which can be captured by considering the covariance structure of the genetic effects among tissues. 

This lends itself to a mixture model, in which  we assume all the gene-snp pairs arise from a mixture of a finite number of multivariate normal (MVN) distribution, each characterized by the covariance matrix from which the vector of effects is though to arise. For each of $\textbf{J}$ gene-snp pairs, we observe an R dimensional vector of standardized effect sizes %$\bm{\hat{b}}$
and their standard error and assume that these effects descend from some true effect size $\bm{b}$. 



 \begin{equation}
  \bm{b_{j}} | \bm{\pi},\bf{U} \sim \sum_{k,l} \pi_{k,l} \;{\it N}_R(\bm{0}, \omega_l U_{k})
  \label{eqn:mixprior}
\end{equation}

Here, the covariance matrix $U_{k}$ captures the particular patterns of sharing - variation in effect sizes within and between tissues, while $\omega_{l}$ determines the scale of each pattern - the magnitude of the effect size. The relationship among the distinct entries of this matrix allows us to specify effects which may be consistently larger in some tissues relative to others and may possess variable correlation among different pairs of tissues (e.g., see Figure \ref{fig:Patterns}, right panel, for converse relationships among effects in tissue one and two for SNPs of green opposed to blue class). Thus we also recognize that while two eQTL may obey a similar pattern or shape, the absolute scale may vary. For example, two eQTL may both have strong correlation between tissues one and two with consistently larger effects in two, but the absolute size of the effects may vary between SNPs (Figure \ref{fig:Patterns}, left panel, contrast gene-snp pairs of far left and far right pattern).\newline

%\textbf{Figure: Colorful patterns, caption indicates different patterns of activity across tissues in groups 1-3, same pattern or shape for snps of group 4 but tendency towards small effects}

\newline
\begin{figure}[htbp]
\includegraphics[width=8cm]{Figures/Patterns.pdf}
\caption{\textbf{Modeling Many Patterns of Sharing} Here, we display a variety of patterns of sharing present in the data. eQTLs belonging to the first class have large positively correlated effects that are consistent in sign though heterogeneous in magnitude. eQTL of pattern 2 display weak consistent effects in tissues 1,2,4 that are inversely correlated with strong tissue specific effects in tissue 3, while eQTL of pattern 3 demonstrate strong effects in a select a group of tissues tissue 2 and 3, but quantitatively weaker though correlated in sign tissues 1 and 4. eQTL of the fourth group show patterns similar in `shape' to eQTL in class 1, though their absolute scale is smaller. A model which simply calls QTLs `active' or `inactive' assumes that all effects fall along the x, y or 45 degree (1,1) line: estimating heterogenous effects which may be consistently larger in some tissues relative to others and may possess variable correlation among different pairs of tissues allows us to capture more diverse biological patterns.}
\label{fig:Patterns}
\end{figure}
\newline


\subsection{Previous Approach}
 
Previous work from our lab considered the idea of configuration - i.e., that a tissue was simply `active' or `inactive' in a particular tissues - and thus for R tissues, there were $2^{R}$ possible configurations, which becomes computationally infeasible as $R$ grows.

Furthermore, this considered only the idea that the variance in effect sizes between two tissues was the same across tissues thought to be active and the covariances were also the same among tissues thought to be active in a given `configuration',  and thus failed to incorporate the much richer covariance structure between tissues. For example, many gene SNP pairs might follow a pattern in which it is common to be `active' across all tissues, but some QTL may have consistently larger effects in liver, lung and thyroid while other QTL may possess consistently larger effects in brain tissues and still another class of gene-snp pairs may show consistently quantitatively specific activity in Whole Blood but non-trivial effects in other tissues. 

As a critical innovation on our previous method \cite{flutre_statistical_2013,wen_bayesian_2014} the covariance matrices used in our method contain distinct diagonal and off-diagonal elements which reflect data-specific patterns of variation within and covariance between subgroups (tissues). This captures the variation in effect sizes within and between subgroups better than restricting effects to simply `shared' or `unshared' between subgroups. 
 
Because we can't know the `true covariance matrix' for each gene-snp pair, we aim to assemble a list which sufficiently captures the various patterns, and then `learn' the relative proportions of each pattern of sharing from the data. One can now model each vector of effect sizes $\bm{b_{j}}$ each as arising from a mixture that captures all the covariance patterns (Equation \ref{eqn:mixprior}).

The primary novelty of this approach is {\it to estimate this multivariate posterior distribution on the effect size in a data-sensitive way} - i.e., using the mixture model to capture information about the covariance structure among subgroups (here, tissues). We deem this model `hierarchical' because these prevailing patterns of activity are learned from the larger dataset - e.g., a large, random set of gene-snp pairs - and influence our inference about a given gene-snp pair `$\bf{j}$'. Thus we might identify a situation in which it is common to have large effects in certain tissues and not others, and thus if a given observed gene-snp pair demonstrates a small effect in one of the `off issues', we might be inclined to conclude that it is indeed a member of this particular class and shrink the small effect in this tissue accordingly without adjusting the more `active tissues' similarly. However, if we observe the the same small effect in a setting in which `similar tissues' have large effects, we might `shrink' this effect size less, due to our high prior belief in the SNP's effectiveness garnered from adjacent tissues. Thus we deem this method `Adaptive Shrinkage' because the appropriate amount of shrinkage is learned from the overall dataset. Critically, our method is dually adaptive, in the sense that we learn the relative abundance of effect sizes and directions from the overarching data set: thus observed effects are nudged towards prevailing patterns and sizes, according to the learned proportions of each.

Because our prior belief in consistency is strong in this particular dataset, we identify many more `significant associations'  in settings where perhaps the observed univariate statistic in one tissue is small but otherwise large in additional tissues, nudging these effects towards something more consistent. This is in contrast to a univariate shrinkage approach, in which all effects of the same size would be `shrunk' equivalently, due to lack of information garnered from adjacent tissues. 

In fact, shrinkage towards $\bm{0}$ of small effects is a result, not a necessity - since the majority of the prior weight is on small $\omega$ components which emphasize components with small prior variance of the effect size $\textbm{b}$, many of the modest observed effects will be smoothed or shrunk towards the prior mean, $\bm{0}$. 

An additional novelty is that in learning about the effect size of a gene-snp pair in each tissue, we can make statements about the degree of heterogeneity present in the data-set- that is the proportion of SNPs who exhibit effects of a different sign or great variety in magnitude, or conversely, the degree of homogeneity should these phenomena be rare. Thus we offer an additional description to eQTL analysis: the degree of heterogeneity across multiple subgroups in both sign and magnitude, by qualifying a SNP by the similarity in size of its effects across subgroups. 